### DISCUSSION TITLE
Style customization

### DISCUSSION THREAD

**Discussion by JulesLassara** [GSI-CS-CO/chart-fx#515] [General]
Hello,

By using chart-fx (thank you again for this lib), I was looking to customize my charts. Specifically change the background colour, and change the main line to dotted line. Do you have any idea how to do this? I haven't found any examples.

**Comment by RalphSteinhagen** [comment]
@JulesLassara many of the `Pane`-type styling follows the normal JavaFX CSS. For the `DataSet`'s which are not derived from CSS-stylable nodes you can style the points and line segments via, for example:

```java
  dataSet.addDataStyle(n, "strokeColor=0xEE00EE; strokeDashPattern=5,8,5,16; fillColor=0xEE00EE"); // for a given data point index 'n'
  dataSet.setStyle(""); // for the DataSet as a whole, in particular lines.
```

You may want to have a look at the examples [here](https://github.com/fair-acc/chart-fx/blob/master/chartfx-samples/src/main/java/de/gsi/chart/samples/LabelledMarkerSample.java) and [here](https://github.com/fair-acc/chart-fx/blob/master/chartfx-samples/src/main/java/de/gsi/chart/samples/ScatterAndBubbleRendererSample.java).

In case you want to customise the default colours, styles etc. have a look at the implementation [DefaultRenderColorScheme.java](https://github.com/fair-acc/chart-fx/blob/master/chartfx-chart/src/main/java/de/gsi/chart/renderer/spi/utils/DefaultRenderColorScheme.java).

Hope this helps...

**Reply by LilRonin** [reply]
where all those style attribut come from? i can't find a proper doc for it. I just want to remove the points on the graph, i want a line to be draw

**Reply by RalphSteinhagen** [reply]
@LilRoning if you mean the names for the style attributes like `strokeColor`, `strokeDashPattern`, etc., they are largely defined in [XYChartCss.java)](https://github.com/fair-acc/chart-fx/blob/master/chartfx-chart/src/main/java/de/gsi/chart/XYChartCss.java).

Hope this helps.

**Reply by LilRonin** [reply]
Yes thank you

**Reply by LilRonin** [reply]
Hello again how can i change background color? like i tried to change style of the stackpane where the graph is put and the xy chart itself but it doesn't work, same goes for changing grid line color
